### PR TITLE
Observe profile changes in conversation view.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/Profiles/OWSProfileManager.h [hunk 1]
```diff
@@ -8,6 +8,9 @@ NS_ASSUME_NONNULL_BEGIN
 
 extern NSString *const kNSNotificationName_LocalProfileDidChange;
 extern NSString *const kNSNotificationName_OtherUsersProfileDidChange;
+extern NSString *const kNSNotificationName_ProfileWhitelistDidChange;
+extern NSString *const kNSNotificationName_ProfileRecipientId;
+extern NSString *const kNSNotificationName_ProfileGroupId;
```
Comment (michaelkirk) [inline Signal/src/Profiles/OWSProfileManager.h]: naming nit: These last two aren't notification names, they are dictionary keys.

File: Signal/src/Profiles/OWSProfileManager.m [hunk 2]
```diff
@@ -660,12 +691,30 @@ - (void)addGroupIdToProfileWhitelist:(NSData *)groupId
...
+        @synchronized(self)
+        {
+            NSString *groupIdKey = [groupId hexadecimalString];
+
+            // We just consult the lazy cache, not the db.
+            if (self.groupProfileWhitelistCache[groupIdKey]) {
+                return;
+            }
+
+            [self.dbConnection setObject:@(1)
```
Comment (michaelkirk) [inline Signal/src/Profiles/OWSProfileManager.m]: `[setBool:forKey:inCollection:]`

File: Signal/src/Profiles/OWSProfileManager.h (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/Profiles/OWSProfileManager.h]: Done.

File: Signal/src/Profiles/OWSProfileManager.m (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/Profiles/OWSProfileManager.m]: Done.

**Pull request merged**